zerver/tests/test_signup.py

```diff
@@ -609,3 +609,52 @@ def test_create_realm(self):
 
             result = self.client.get(result["Location"])
             self.assertIn("You're the first one here!", result.content)
+
+    def test_create_realm_with_subdomain(self):
+        # type: () -> None
+        username = "user1"
+        password = "test"
+        domain = "test.com"
+        email = "user1@test.com"
+        subdomain = "test"
+        realm_name = "Test"
+
+        # Make sure the realm does not exist
+        self.assertIsNone(get_realm(domain))
+        with self.settings(REALMS_HAVE_SUBDOMAINS=True), self.settings(OPEN_REALM_CREATION=True):
```

Comment:
This is the use case. The test won't pass as the value of OPEN_REALM_CREATION will remain False in forms.py.